assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/clip/clip_embeddings_mlflow_wrapper.py

```diff
@@ -0,0 +1,195 @@
+        """
+        Load a MLflow model with pyfunc.load_model().
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        if self._task_type == Tasks.IMAGE_TEXT_EMBEDDINGS.value:
+            try:
+                model_dir = context.artifacts[MLflowLiterals.MODEL_DIR]
+                self._processor = AutoProcessor.from_pretrained(model_dir)
+                self._model = AutoModelForZeroShotImageClassification.from_pretrained(model_dir)
+                self._device = get_current_device()
+                self._model.to(self._device)
+
+                print("Model loaded successfully")
+            except Exception as e:
+                print("Failed to load the the model.")
+                print(e)
+                raise
+        else:
+            raise ValueError(f"invalid task type {self._task_type}")
+
+    def predict(self, context: mlflow.pyfunc.PythonModelContext, input_data: pd.DataFrame) -> pd.DataFrame:
+        """Perform inference on the input data.
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        :param input_data: Input images and text for feature embeddings.
+        :type input_data: Pandas DataFrame with a first column name ["image"] of images where each
+        image is in base64 String format, and second column name ["text"] containing a string. The following cases
+        are supported:
+        - all rows in image column are populated with valid values and text column only contains empty string,
+        - all rows in text column are populated with valid values and image column only contains empty string,
+        - all rows in both columns are populated with valid values
+        :return: Output of inferencing
+        :rtype: Pandas DataFrame with columns "image_features" and/or "text_features"
+        """
+        has_images, has_text = self.validate_input(input_data)
+
+        if has_images:
+            # Decode the base64 image column
+            decoded_images = input_data.loc[
+                :, [MLflowSchemaLiterals.INPUT_COLUMN_IMAGE]
+            ].apply(axis=1, func=process_image_pandas_series)
+
+        if has_text:
+            text_list = input_data[MLflowSchemaLiterals.INPUT_COLUMN_TEXT].tolist()
+        else:
+            text_list = None
+
+        with tempfile.TemporaryDirectory() as tmp_output_dir:
+            if has_images:
+                image_path_list = (
+                    decoded_images.iloc[:, 0]
+                    .map(lambda row: create_temp_file(row, tmp_output_dir)[0])
+                    .tolist()
+                )
+            else:
+                image_path_list = None
+            image_features, text_features = self.run_inference_batch(
+                processor=self._processor,
+                model=self._model,
```

Comment:
No need to pass the processor or the model to `run_inference_batch()`, as it has access to them via self.